frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -199,21 +200,23 @@ class PopupCreateFile extends React.Component {
 
     Promise.all(state.fileToUploadList.map(async (file) =>
       this.buildUploadedFileResponse(file)
-    )).then(uploadedFileResponseList => {
-      this.handleAllFileUploadEnd(uploadedFileResponseList.map((uploadedFileResponse) => this.handleFileUploadEnd(uploadedFileResponse.xhr, uploadedFileResponse.file)))
+    )).then(async uploadedFileResponseList => {
+      const fileUploadList = await Promise.all(uploadedFileResponseList.map(async (uploadedFileResponse) => this.handleFileUploadEnd(uploadedFileResponse.xhr, uploadedFileResponse.file)))
+      this.handleAllFileUploadEnd(fileUploadList)
     }).catch(() => {
       this.sendGlobalFlashMessage(props.t('Error while creating file'))
       this.handleClose(false)
     })
   }
 
-  handleFileUploadEnd = (xhr, file) => {
-    const { props } = this
+  handleFileUploadEnd = async (xhr, file) => {
+    const { state, props } = this
 
     const filePosted = new File([file], file.name)
     switch (xhr.status) {
       case 200:
         const jsonResult200 = JSON.parse(xhr.responseText)
+        await putMyselfFileRead(state.config.apiUrl, state.workspaceId, jsonResult200.content_id)
```

Comment:
is the await necessary here ?

Reply:
Yes, because we need to wait that each files is put to read before refresh the list of contents, (in `handleAllFileUploadEnd()`